src/components/pages/Authentication/Login.jsx

```diff
@@ -0,0 +1,176 @@
+import { useStore } from "../../../store/app-store-context";
+import { emailValidate, passwordValidate } from "../../../util";
+import "./auth.css";
+
+const Login = () => {
+  const getEmail = () => JSON.parse(localStorage.getItem("rememberme")) || "";
+
+  const [email, setEmail] = useState(getEmail);
+  const [password, setPassword] = useState("");
+
+  const [isEmailValid, setIsEmailVaild] = useState(false);
+  const [isPasswordValid, setIsPasswordVaild] = useState(false);
+
+  const [rememberMe, setRememberMe] = useState(false);
+  const [isFormValid, setFormValid] = useState(false);
+  const { isLoggedIn, actions } = useStore();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      navigate("/");
+    }
+  }, [isLoggedIn]);
+
+  useEffect(() => {
+    setFormValid(isEmailValid && isPasswordValid);
+  }, [isEmailValid, isPasswordValid]);
+
+  const loginHandler = (e) => {
+    e.preventDefault();
+    if (!isFormValid) {
+      return;
+    }
+    actions.login({ email, password });
+    if (rememberMe) {
+      localStorage.setItem("rememberme", JSON.stringify(email));
+    } else {
+      localStorage.removeItem("rememberme");
+    }
+    setEmail("");
+    setPassword("");
+  };
+
+  const onEmailChange = (value) => {
+    const isEmailValid = emailValidate(value);
+    setIsEmailVaild(isEmailValid);
+  };
+
+  const onPasswordChange = (value) => {
+    const isPasswordValid = passwordValidate(value);
+    setIsPasswordVaild(isPasswordValid);
+  };
+
+  const confirmationBuilder = (label, handler, checked) => {
+    return (
+      <div className="flex-row flex-jc">
+        <input
+          id="confirmation"
+          type="checkbox"
+          value={checked}
+          onClick={handler}
+        />
+        <label className="confirmation font-normal" htmlFor="confirmation">
+          {label}
+        </label>
+      </div>
+    );
+  };
+  const onClickRememeberMe = (e) => {
+    setRememberMe(e.target.checked);
+  };
+
+  const getConfirmationSection = () => {
+    const confirmationText = "Remember me";
+    const handler = onClickRememeberMe;
+    const checked = rememberMe;
+    return confirmationBuilder(confirmationText, handler, checked);
+  };
+  const userDetails = (
+    <>
+      <div className="form-group">
+        <label className="label font-medium" htmlFor="username" name="username">
+          Username :
+        </label>
+        <input
+          id="username"
+          className="input"
+          type="text"
+          value={email}
+          placeholder="tony.stark@gmail.com"
+          autoComplete="off"
+          onInput={(e) => setEmail(e.target.value)}
+          onBlur={(e) => onEmailChange(e.target.value)}
+        />
+        {!isEmailValid && !!email && (
+          <span className="inline-block validation__msg input__error">
+            <i className="fas fa-exclamation-triangle"></i>
+            <span className="px-3">Email is invalid</span>
```

Comment:
this seems to bug out if i copy paste an email id, do check

Reply:
yeah copy paste is not working. I tried with onChange but same result. that is an open issue I am yet to figure out the solution.